update-repo and go1.16 still rely on go.sum

<!-- Please answer these questions before submitting your issue. Thanks! -->

### What version of gazelle are you using?

gazelle head master 5b8616dbb7dad825c61ffbb12ea9f75622568657

<!-- Check bazel_gazelle in WORKSPACE if you're not sure -->

### What version of rules_go are you using?

rules_go v0.27.0

<!-- Check io_bazel_rules_go in WORKSPACE if you're not sure -->

### What version of Bazel are you using?

<!-- Run "bazel version" to find out -->

### Does this issue reproduce with the latest releases of all the above?

Yes.

### What operating system and processor architecture are you using?

MacOS Catalina

### What did you do?

gazelle update-repos -from_file=go.mod -to_macro=go_repositories.bzl%go_repositories -prune=true -build_directives=build_naming_convention=go_default_library


<!--
If possible, provide a minimal recipe for reproducing the error.
-->

### What did you expect to see?

Successful execution of gazelle update-repos. We shouldn't rely on go.sum

### What did you see instead?

`gazelle: running '/private/var/tmp/_bazel_aberardino/628a60d6b6ad744cbde16d3a986e35c8/external/go_sdk/bin/go /private/var/tmp/_bazel_aberardino/628a60d6b6ad744cbde16d3a986e35c8/external/go_sdk/bin/go list -mod=readonly -m -json all': go: github.com/ahmdrz/goinsta/v2@v2.4.5: missing go.sum entry; to add it:
	go mod download github.com/ahmdrz/goinsta/v2
make: *** [godeps] Error 1`

--- a/language/go/modules.go
+++ b/language/go/modules.go
@@ -151,7 +151,7 @@ func importReposFromModules(args language.ImportReposArgs) language.ImportReposR
 
 // goListModules invokes "go list" in a directory containing a go.mod file.
 var goListModules = func(dir string) ([]byte, error) {
-	return runGoCommandForOutput(dir, "list", "-mod=readonly", "-m", "-json", "all")
+	return runGoCommandForOutput(dir, "list", "-mod=readonly", "-e", "-m", "-json", "all")
 }
 
 // goModDownload invokes "go mod download" in a directory containing a

